### PR TITLE
docs(app-vite): improve configuration sections

### LINKED ISSUE #15892: Mention TS aliases in app-webpack to app-vite migration guide
### What happened?

Following the migration guide I edited files and copied source files. I only build SPA.
First, there is no easy way to get the path aliases from typescript into vite within the quasar.config.js.
Second, vite is unable to load the src/css/app.sass and throws an error in the browser that the file was blocked due to disallowed MIME Type "".
When adding an alias it now can be found.
Sadly, that does not count for the src/App.vue. I get the same error here which is a showstopper

### What did you expect to happen?

The migration guide towards app-vue would be working with typescript, path aliases

### Reproduction URL

unable to provice one

### How to reproduce?

Following the migration guide steps and start quasar dev

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Quasar CLI Commands/Configuration (@quasar/cli | @quasar/app-webpack | @quasar/app-vite), Quasar Extras (@quasar/extras), TypeScript Support, SPA Mode

### Platforms/Browsers

Firefox

### Quasar info output

```Shell
Operating System - Windows_NT(10.0.19044) - win32/x64
NodeJs - 18.14.0
Global packages
  NPM - 9.3.1
  yarn - 1.22.4
  @quasar/cli - undefined
  @quasar/icongenie - Not installed
  cordova - Not installed
Important local packages
  quasar - 2.11.10 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.4.3 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.16.3 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.47 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.1.6
  pinia - Not installed
  vuex - Not installed
  vite - 3.2.6 -- Native-ESM powered web dev build tool
  eslint - 8.39.0 -- An AST-based pattern checker for JavaScript.
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed
Quasar App Extensions
  @quasar/quasar-app-extension-apollo - 2.0.0-beta.5 -- A Quasar app extension to add GraphQL support using Apollo Client.
  @quasar/quasar-app-extension-testing-unit-jest - 3.0.0-alpha.10 -- A Quasar App Extension for running Jest tests
  @quasar/quasar-app-extension-testing - 2.1.0 -- A Quasar App Extension for managing Test Harnesses
  @quasar/quasar-app-extension-testing-e2e-cypress - 5.1.0 -- A Quasar App Extension for Cypress e2e
```

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yusufkandemir**
**What kind of change does this PR introduce?**

- Documentation

**Does this PR introduce a breaking change?**

- No

**The PR fulfills these requirements:**

- It's submitted to the `dev` branch (or `v[X]` branch)

**Other information:**
Fixes #15892 but contains much more.

### CODE REVIEW

File: app-vite/types/configuration/framework-conf.d.ts [hunk 1]
```diff
@@ -13,40 +15,91 @@ type SerializableConfiguration<T> = {
...
    * @default false
    */
   devTreeshaking?: boolean;
 
-  components?: (keyof QuasarPluginOptions["components"])[];
-  directives?: (keyof QuasarPluginOptions["directives"])[];
-  plugins?: (keyof QuasarPluginOptions["plugins"])[];
+  /**
+   * Quasar will auto import components based on your usage.
+   * But, in case you have a special case, you can manually specify Quasar components to be available everywhere.
```
Comment (smolinari) [inline app-vite/types/configuration/framework-conf.d.ts:78]: Being picky, but probably better.

* But, should you have a special case,

File: app-vite/types/configuration/framework-conf.d.ts [hunk 2]
```diff
@@ -13,40 +15,91 @@ type SerializableConfiguration<T> = {
...
+   *
+   * An example case would be having custom component definitions with plain string templates, inside .js or .ts files,
```
Comment (smolinari) [inline app-vite/types/configuration/framework-conf.d.ts:80]: > An example case would be having custom component definitions with plain string templates, inside .js or .ts files,

An example case would be to have custom component definitions with plain string templates inside .js or .ts files,

File: app-vite/types/configuration/framework-conf.d.ts [hunk 3]
```diff
@@ -13,40 +15,91 @@ type SerializableConfiguration<T> = {
...
+   * in which you are using Quasar components (e.g. q-avatar).
+   *
+   * Another example would be that dynamically rendering components depending on an API response or similar (e.g. in a CMS),
```
Comment (smolinari) [inline app-vite/types/configuration/framework-conf.d.ts:83]: Another example would be to have components dynamically rendered depending on..

File: app-vite/types/configuration/framework-conf.d.ts [hunk 4]
```diff
@@ -13,40 +15,91 @@ type SerializableConfiguration<T> = {
...
+   * something like `<component :is="dynamicName">` where `dynamicName` is a string that matches a Quasar component name.
+   *
+   * @example ['QAvatar', 'QChip']
+   */
+  components?: (keyof QuasarComponents)[];
+  /**
+   * Quasar will auto import directives based on your usage.
+   * But, in case you have a special case, you can manually specify Quasar directives to be available everywhere.
```
Comment (smolinari) [inline app-vite/types/configuration/framework-conf.d.ts:91]: same as above. :)

File: app-vite/types/configuration/framework-conf.d.ts [hunk 5]
```diff
@@ -13,40 +15,91 @@ type SerializableConfiguration<T> = {
...
+   *
+   * An example case would be having custom component definitions with plain string templates, inside .js or .ts files,
```
Comment (smolinari) [inline app-vite/types/configuration/framework-conf.d.ts:93]: An example case would be to have custom component definitions ...

File: docs/src/pages/quasar-cli-vite/convert-to-quasar-cli-with-vite.md [hunk 6]
```diff
@@ -72,9 +72,13 @@ import MyComponent from './MyComponent.vue'
 
 There are property changes in `build`, `devServer`, and all Quasar Modes (pwa, ssr, etc). The props are detailed in the [quasar.config file](/quasar-cli-vite/quasar-config-file) page. You will have to manually port your configuration to the Quasar CLI with Vite architecture.
 
+#### TypeScript Aliases
+
+If you are using TypeScript and defined custom path aliases in `tsconfig.json > compilerOptions > paths`, they will no longer be processed automatically. See [Folder aliases | Handling Vite](quasar-cli-vite/handling-vite#folder-aliases) for available ways to handle this.
```
Comment (smolinari) [inline docs/src/pages/quasar-cli-vite/convert-to-quasar-cli-with-vite.md:77]: If you are using TypeScript and have defined....

File: docs/src/pages/quasar-cli-vite/quasar-config-file.md [hunk 7]
```diff
@@ -230,54 +244,106 @@ extras?: (QuasarIconSets | QuasarFonts)[];
...
+   *
+   * @default false
+   */
+  devTreeshaking?: boolean;
 
   /**
-   * Quasar plugins.
-   * @example [ 'Notify', 'Loading', 'Meta', 'AppFullscreen' ]
+   * Quasar will auto import components based on your usage.
+   * But, in case you have a special case, you can manually specify Quasar components to be available everywhere.
```
Comment (smolinari) [inline docs/src/pages/quasar-cli-vite/quasar-config-file.md:312]: Same as above

File: docs/src/pages/quasar-cli-vite/quasar-config-file.md [hunk 8]
```diff
@@ -230,54 +244,106 @@ extras?: (QuasarIconSets | QuasarFonts)[];
...
+   *
+   * An example case would be having custom component definitions with plain string templates, inside .js or .ts files,
```
Comment (smolinari) [inline docs/src/pages/quasar-cli-vite/quasar-config-file.md:314]: Same as above. :)

File: docs/src/pages/quasar-cli-vite/quasar-config-file.md [hunk 9]
```diff
@@ -230,54 +244,106 @@ extras?: (QuasarIconSets | QuasarFonts)[];
...
+   * in which you are using Quasar components (e.g. q-avatar).
+   *
+   * Another example would be that dynamically rendering components depending on an API response or similar (e.g. in a CMS),
```
Comment (smolinari) [inline docs/src/pages/quasar-cli-vite/quasar-config-file.md:317]: Same as above.

File: docs/src/pages/quasar-cli-vite/quasar-config-file.md [hunk 10]
```diff
@@ -230,54 +244,106 @@ extras?: (QuasarIconSets | QuasarFonts)[];
...
+   * something like `<component :is="dynamicName">` where `dynamicName` is a string that matches a Quasar component name.
+   *
+   * @example ['QAvatar', 'QChip']
+   */
+  components?: (keyof QuasarComponents)[];
+  /**
+   * Quasar will auto import directives based on your usage.
+   * But, in case you have a special case, you can manually specify Quasar directives to be available everywhere.
```
Comment (smolinari) [inline docs/src/pages/quasar-cli-vite/quasar-config-file.md:325]: Same as above. I'll stop here. You should have the gist by now. 😁

**Comment by rstoenescu** [comment]
@smolinari I just saw your comments on this one. It got merged already. Can you do another PR pls? Will release q/apps after it.

**Comment by smolinari** [comment]
@rstoenescu - They were just picky grammatical improvements. It's generally ok as is. 

Scott

**Pull request merged**